autotest/gcore/vrt_read.py

```diff
@@ -1442,25 +1442,27 @@ def test_vrt_invalid_source_band():
 @gdaltest.enable_exceptions()
 def test_vrt_protocol():
 
+    files_opened_start = gdaltest.get_opened_files()
+
     with pytest.raises(Exception):
-        assert not gdal.Open("vrt://")
```

Comment:
`AssertionError` subclasses `Exception`, so if `gdal.Open` returns a dataset then our assertion will fail and `pytest.raises` will find what it was looking for.